tests/Application/Actions/Donations/UpdateTest.php

```diff
@@ -1218,6 +1219,210 @@ public function testAddDataHitsAlreadyCapturedStripeExceptionWithFeeChange(): vo
         ], $payloadArray);
     }
 
+    /**
+     * Success on retry.
+     */
+    public function testAddDataHitsStripeLockExceptionOnce(): void
+    {
+        $app = $this->getAppInstance();
+        /** @var Container $container */
+        $container = $app->getContainer();
+
+        $donation = $this->getTestDonation();
+        $donation->setDonorCountryCode('US');
+        $donation->setCurrencyCode('USD');
+        $donation->setTipAmount('3.21');
+        $donation->setGiftAid(true);
+        $donation->setTipGiftAid(false);
+        $donation->setDonorHomeAddressLine1('99 Updated St');
+        $donation->setDonorHomePostcode('X1 1XY');
+        $donation->setDonorFirstName('Saul');
+        $donation->setDonorLastName('Williams');
+        $donation->setDonorEmailAddress('saul@example.com');
+        $donation->setTbgComms(true);
+        $donation->setCharityComms(false);
+        $donation->setChampionComms(false);
+        $donation->setDonorBillingAddress('Y1 1YX');
+
+        $donationRepoProphecy = $this->prophesize(DonationRepository::class);
+        $donationRepoProphecy
+            ->findOneBy(['uuid' => '12345678-1234-1234-1234-1234567890ab'])
+            ->willReturn($this->getTestDonation()) // Get a new mock object so DB has old values.
+            ->shouldBeCalledOnce();
```

Comment:
shouldBeCalledOnce is probably unnecessary - generally I expect we want [Mocks for Commands, Stubs for Queries](https://blog.ploeh.dk/2013/10/23/mocks-for-commands-stubs-for-queries/l)
```suggestion
            ->willReturn($this->getTestDonation()); // Get a new mock object so DB has old values.
```

Reply:
Maybe we can discuss it if it comes up again on another PR when we're less pushed for time. I think the mocks for commands, stubs for queries principle is very helpful when writing or editing tests, but it's unlikely to be worthwhile to edit old tests purely for the sake of following the principle.